sql/expression/function/uuid.go

```diff
@@ -0,0 +1,485 @@
+//
+// The MAC address of an interface is taken into account only on FreeBSD, Linux, and Windows. On other operating systems,
+// MySQL uses a randomly generated 48-bit number.
+// https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid
+
+type UUIDFunc struct{}
+
+var _ sql.FunctionExpression = &UUIDFunc{}
+
+func NewUUIDFunc() sql.Expression {
+	return UUIDFunc{}
+}
+
+func (U UUIDFunc) String() string {
+	return "UUID()"
+}
+
+func (U UUIDFunc) Type() sql.Type {
+	return sql.MustCreateStringWithDefaults(sqltypes.VarChar, 36)
+}
+
+func (U UUIDFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	nUUID, err := uuid.NewUUID()
+	if err != nil {
+		return nil, err
+	}
+
+	return nUUID.String(), nil
+}
+
+func (U UUIDFunc) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 0 {
+		return nil, sql.ErrInvalidChildrenNumber.New(U, len(children), 0)
+	}
+
+	return &UUIDFunc{}, nil
+}
+
+func (U UUIDFunc) FunctionName() string {
+	return "uuid"
+}
+
+func (U UUIDFunc) Resolved() bool {
+	return true
+}
+
+// Children returns the children expressions of this expression.
+func (U UUIDFunc) Children() []sql.Expression {
+	return nil
+}
+
+// IsNullable returns whether the expression can be null.
+func (U UUIDFunc) IsNullable() bool {
+	return false
+}
+
+// IS_UUID(string_uuid)
+//
+// Returns 1 if the argument is a valid string-format UUID, 0 if the argument is not a valid UUID, and NULL if the
+// argument is NULL.
+//
+// “Valid” means that the value is in a format that can be parsed. That is, it has the correct length and contains only
+// the permitted characters (hexadecimal digits in any lettercase and, optionally, dashes and curly braces).
+
+type IsUUID struct {
+	child sql.Expression
+}
+
+var _ sql.FunctionExpression = &IsUUID{}
+
+func NewIsUUID(arg sql.Expression) sql.Expression {
+	return &IsUUID{child: arg}
+}
+
+func (u *IsUUID) String() string {
+	return fmt.Sprintf("IS_UUID(%s)", u.child)
+}
+
+func (u *IsUUID) Type() sql.Type {
+	return sql.Int8
+}
+
+func (u *IsUUID) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	str, err := u.child.Eval(ctx, row)
+	if err != nil {
+		return 0, err
+	}
+
+	if str == nil {
+		return nil, nil
+	}
+
+	switch str := str.(type) {
+	case string:
+		_, err := uuid.Parse(str)
+		if err != nil {
+			return int8(0), nil
+		}
+
+		return int8(1), nil
+	case []byte:
+		_, err := uuid.ParseBytes(str)
+		if err != nil {
+			return int8(0), nil
+		}
+
+		return int8(1), nil
+	default:
+		return int8(0), nil
+	}
+}
+
+func (u *IsUUID) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 1 {
+		return nil, sql.ErrInvalidChildrenNumber.New(u, len(children), 1)
+	}
+
+	return &IsUUID{child: children[0]}, nil
+}
+
+func (u *IsUUID) FunctionName() string {
+	return "is_uuid"
+}
+
+func (u *IsUUID) Resolved() bool {
+	return u.child.Resolved()
```

Comment:
This was the source of your bug I fixed. All the other functions you added have the same problem.

You should write failing engine tests that catch the bug, for every function. Then fix it.